src/GraphQL/Utility/FileUpload.php

```
@@ -0,0 +1,254 @@
+    }
+  }
+
+  /**
+   * Gets violations.
+   *
+   * @return array
+   *   List of violations.
+   */
+  public function getViolations() {
+    return $this->violations;
+  }
+
+  /**
+   * Gets max upload size.
+   *
+   * @param array $settings
+   *   The file field settings.
+   *
+   * @return int
+   *   Max upload size.
+   */
+  protected function getMaxUploadSize(array $settings) {
+    // Cap the upload size according to the PHP limit.
+    $max_filesize = Bytes::toInt(Environment::getUploadMaxSize());
+    if (!empty($settings['max_filesize'])) {
+      $max_filesize = min($max_filesize, Bytes::toInt($settings['max_filesize']));
+    }
+    return $max_filesize;
+  }
+
+  /**
+   * Retrieves the upload validators for a file field.
+   *
+   * @param array $settings
+   *   The file field settings.
+   *
+   * @return array
+   *   List of file validators.
+   */
+  protected function getUploadValidators(array $settings) {
+    // Validate name length.
+    $validators = [
+      'file_validate_name_length' => [],
+    ];
+
+    // There is always a file size limit due to the PHP server limit.
+    $validators['file_validate_size'] = [$this->getMaxUploadSize($settings)];
+
+    // Add the extension check if necessary.
+    if (!empty($settings['file_extensions'])) {
+      $validators['file_validate_extensions'] = [$settings['file_extensions']];
+    }
+
+    return $validators;
+  }
+
+  /**
+   * Create a temporary file and send back the newly created entity.
+   *
+   * Based on several file upload handlers, see
+   * _file_save_upload_single()
+   * \Drupal\file\Plugin\Field\FieldType\FileItem
+   * \Drupal\file\Plugin\rest\resource\FileUploadResource.
+   *
+   * @param \Symfony\Component\HttpFoundation\File\UploadedFile $file
+   *   The file entity to upload.
+   * @param array $settings
+   *   Fle settings as specified in regular file field config.
+   *
+   * @return \Drupal\file\FileInterface|bool
```

Instead of returning FALSE i'd vote for returning NULL in case of a failure.